packages/framework-integration-tests/integration/provider-unaware/end-to-end/read-models.integration.ts

```diff
@@ -795,6 +795,119 @@ describe('Read models end-to-end tests', () => {
         expect(cartData[0].payment.id).to.be.eq(mockPaymentId)
       })
 
+      it('should retrieve a list of carts when filter by regex', async () => {
+        if (process.env.TESTED_PROVIDER !== 'AZURE' && process.env.TESTED_PROVIDER !== 'LOCAL') {
+          console.log('****************** Warning **********************')
+          console.log('Only Azure and Local provider implement the regex filter. Skipping')
+          console.log('*************************************************')
+          return
+        }
+        const filter = {
+          and: [
+            {
+              id: { eq: mockCartId },
+            },
+            {
+              shippingAddress: {
+                firstName: {
+                  regex: `^${mockAddress.firstName.at(0)}.*`,
+                },
+              },
+            },
+          ],
+        }
+        const queryResult = await waitForIt(
```

Comment:
Super nit-picky, why not reuse in the tests a function to get the `ListCartReadModels` passing only the different filters `regex` or `iRegex`?

Reply:
I agree here. It would make super easy to test this more thoroughly by just changing the regex expression